src/network_service/sending_targets_cache.rs

```diff
@@ -6,28 +6,39 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::{quic_p2p::Token, ConnectionInfo};
+use crate::{
+    quic_p2p::Token,
+    time::{Duration, Instant},
+    ConnectionInfo,
+};
 use log::LogLevel;
 use std::{
     collections::{hash_map::Entry, HashMap},
     net::SocketAddr,
 };
 
-const MAX_RESENDS: u8 = 3;
+/// Time interval during which we attempt to resend a message to a given target before giving up.
+pub const RESEND_TIMEOUT: Duration = Duration::from_secs(30);
```

Comment:
In my mind I'd think I'd probably have both. In particular, lets say you start with node A1. then you'll try node A2, A3, A4, A5, then your back to A1, then time out is gone.

I guess I had more something like MAX_TRY, but if it reach it too fast, continue until timeout pass at least.

Reply:
Or maybe we should somehow pause the timeout while a target is in the failed state. I'll see which approach is better.

Reply:
I think if we are trying a node, we should try a few times with a pause between each try. However, rather than wait on that node, we should just move on to the next one. So we try them all quickly then a short pause and go around again, we can do this twice (so 3 attempts per node in total)  and if it fails then we need to error.